### PR TITLE
Add a test suite

### PR DESCRIPTION
**Pull request by alexblackie**
This pulls in RSpec and begins writing unit tests for some of this.

Also we add TravisCI to run the test suite on PRs.

### CODE REVIEW

File: spec/fetch_spec.rb [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    end
+
+  end
+
+  describe "#add_videos_for_channel" do
+    let(:channel) do
+    end
+    subject { service.add_videos_for_channel(channel) }
+
+    it "creates video records" do
```
Comment (jhawthorn) [inline spec/fetch_spec.rb]: missing test body?

**Review by jhawthorn — CHANGES_REQUESTED** [review]
One test missing?

File: spec/fetch_spec.rb (hunk 1 above)
Comment (alexblackie) [reply spec/fetch_spec.rb]: lol

**Review by jhawthorn — APPROVED** [review]
lgtm

**Pull request merged**